File: werewolf-simulator/app/models/game.py
from pydantic import BaseModel, Field

from app.models.action import NightAction
from app.models.player import Player
from app.models.role import GamePhase, Winner


class GameState(BaseModel):
    game_id: str
    day_number: int = 1
    phase: GamePhase = GamePhase.NIGHT_WOLF
    players: list[Player]
    night_action: NightAction = Field(default_factory=NightAction)
    votes: dict[str, str] = Field(default_factory=dict)
    speeches: dict[str, str] = Field(default_factory=dict)
    speech_cursor: int = 0
    public_logs: list[str] = Field(default_factory=list)
    debug_logs: list[str] = Field(default_factory=list)
    winner: Winner | None = None
    witch_antidote_available: bool = True
    witch_poison_available: bool = True
    pending_night_deaths: list[str] = Field(default_factory=list)
    pending_exile: str | None = None

    def get_player(self, player_id: str) -> Player | None:
        return next((player for player in self.players if player.player_id == player_id), None)

    def alive_players(self) -> list[Player]:
        return [player for player in self.players if player.is_alive]

File: werewolf-simulator/frontend/src/types/game.ts
export type Role = "WEREWOLF" | "SEER" | "WITCH" | "VILLAGER";

export type GamePhase =
  | "NIGHT_WOLF"
  | "NIGHT_SEER"
  | "NIGHT_WITCH"
  | "DAY_ANNOUNCEMENT"
  | "DAY_SPEECH"
  | "DAY_VOTE"
  | "EXILE"
  | "GAME_OVER";

export type Winner = "GOOD" | "WEREWOLF";

export interface Player {
  player_id: string;
  nickname: string;
  is_alive: boolean;
  seat_number: number;
  is_ai: boolean;
  role?: Role;
}

export interface GameState {
  game_id: string;
  day_number: number;
  phase: GamePhase;
  players: Player[];
  votes: Record<string, string>;
  public_logs: string[];
  debug_logs?: string[];
  winner: Winner | null;
}

export interface StepResponse {
  game_id: string;
  phase: GamePhase;
  new_public_logs: string[];
  is_over: boolean;
  winner: Winner | null;
}

export interface RunResponse {
  game_id: string;
  winner: Winner | null;
  public_logs: string[];
}

export interface LogsResponse {
  public_logs: string[];
  debug_logs?: string[];
}

export interface HealthResponse {
  status: string;
}
